### PR TITLE
Feat/go gen ci check

### PR DESCRIPTION
**Pull request by 3u13r**
### Proposed change(s)

Add a CI test which checks if running `go generate ./...` leads to changes. 

Thanks to @katexochen for the help.

### CODE REVIEW

File: .github/actions/install_docgen/action.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  steps:
+    - name: Checkout talos
+      uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # tag=v3.1.0
+      with:
+        fetch-depth: 0
+        repository: "siderolabs/talos"
+        ref: "v1.2.6"
+        path: talos
+
+    - name: Remove go.work
```
Comment (katexochen) [inline .github/actions/install_docgen/action.yml:15]: Maybe a good place to leave a comment that this is required for building docgen.

File: .github/actions/install_docgen/action.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      shell: bash
+      working-directory: talos
+      run: rm go.work
```
Comment (katexochen) [inline .github/actions/install_docgen/action.yml:18]:
```suggestion
      run: rm go.work*
```
Surely not needed, but who knows what future Go tooling will do if there is a lonely go.work.sum...

File: .github/workflows/test-go-generate.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+name: Go generate check
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - "release/**"
+  pull_request:
```
Comment (katexochen) [inline .github/workflows/test-go-generate.yml:9]:
```suggestion
  workflow_dispatch:
  push:
    branches:
      - main
      - "release/**"
    paths:
      - "**.go"
      - "**/go.mod"
      - "**/go.sum"
  pull_request:
    paths:
      - "**.go"
      - "**/go.mod"
      - "**/go.sum"
```

This is only needed on changes in Go files, isn't it?

File: .github/workflows/test-go-generate.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Install stringer
+        shell: bash
+        run: go install golang.org/x/tools/cmd/stringer@latest
+
+      - name: Go generate check
+        shell: bash
+        working-directory: constellation
+        run: |
+          go generate ./...
+          git diff --exit-code
```
Comment (katexochen) [inline .github/workflows/test-go-generate.yml:40]: This should be done with multi-module support.
```suggestion
      - name: Get Go submodules
        id: submods
        shell: bash
        run: |
          mods=$(go list -f '{{.Dir}}' -m | xargs)
          echo "Found mods: $mods"
          echo "submods=${mods}" >> "$GITHUB_OUTPUT"
       
      - name: Go generate check
        shell: bash
        working-directory: constellation
        run: |
          mods=${{ steps.submods.outputs.submods }}
          result=0
          for mod in $mods; do
              (cd $mod; go generate ./...; git diff --exit-code || result=1)
          done
          exit $result
```

Comment (katexochen) [reply .github/workflows/test-go-generate.yml:40]: Note for myself to check how this behaves if a tool referenced in go generate is not installed...

**Review by katexochen — CHANGES_REQUESTED** [review]

File: .github/workflows/test-go-generate.yml (hunk 3 above)
Comment (katexochen) [reply .github/workflows/test-go-generate.yml:9]: Ah, surely this is not the case for the generation of the cilium helm chart...

File: cli/internal/helm/charts/cilium/README.md [hunk 5]
```diff
@@ -37,7 +37,7 @@ $ helm install cilium cilium/cilium --namespace=kube-system
 ```
 
 After Cilium is installed, you can explore the features that Cilium has to
-offer from the [Getting Started Guides page](https://docs.cilium.io/en/stable/gettingstarted/).
+offer from the [Getting Started Guides page](https://docs.cilium.io/en/latest/gettingstarted/).
```
Comment (daniel-weisse) [inline cli/internal/helm/charts/cilium/README.md:40]: This link in the upstream cilium is broken, maybe we should just add it to the ignore list?

Comment (3u13r) [reply cli/internal/helm/charts/cilium/README.md:40]: Already done in .github/workflows/check-links.yml:29

**Pull request merged**